### PR TITLE
Authorization member section

### LINKED ISSUE #397: Authorization implemented - member section
Description of roles in perun: https://is.muni.cz/auth/th/t4sdm/Master_Thesis_final.pdf
Where to find policies what needs to be used: perun-roles.yml in perun repository(backend)
What needs to be done:
In entity-overview hide items that user dont have role to see them - what does that means - eg. - vo - application item - user shouldnt have see that item if he doesnt have the authority to call the method getApplication(example)
Same rules are applicated to side-menu.
Also check the entity description(there could be link into the user detail or facility detail) and you must check if that link should be available.

Then when we go to the eg. vo-groups page we agreed on:
1. button - create/add - this should be easy look on vo-groups button add, I already did an example how to do that
2. button - remove/delete - when checking/unchecking the select checkbox in the table always check if the user have the authority to delete this group and if dont make the delete button disabled and make the tooltip on that button say that he doesnt have the priviledge to delete some of the selected groups.
3. clicking on the detail of the entity - always check on first entity in the list if the user have priviledge to go into the detail of entity. There is example in vo-groups.

Your fake account that you have created use for the testing if your solution works. If that wouldnt work there would be PerunPriviledge exception notification showing up.

### PR DESCRIPTION
**Pull request by bodnara**
Resolves #397 

Added authorization for:
* Overview - items, expiration attribute (authorization for this attr. is hardcoded for now)
* Sidemenu items
* Groups - Add/Remove buttons, routing, Add dialog (display of group)
* Resources - Add button, routing, Add dialog (disabled selection)
* Attributes - attribute list now disables checkbox and input for attributes that user cannot edit, on hover a hint is displayed

### CODE REVIEW

**Review by xkostka2 — CHANGES_REQUESTED** [review]
image

Authorization in Overview, Groups, Applications and Resources pages does not work at all.

**Comment by bodnara** [comment]
> Authorization in Overview, Groups, Applications and Resources pages does not work at all (tested on a member without any admin privileges).

I am not sure, how did you manage to get to those pages as a member without any privileges. Could you give more information on how to recreate this problem? When I tried to access those pages as a member, it was not possible to get there without manually typing the URL. And even when then it worked properly. 
image

**Review by xkureck — CHANGES_REQUESTED** [review]
image
As Vo observer I can get into the member detail but also I see the edit button for expiraton, vo observer doesnt have rights to change it so hide it if the user dont have the rights.
image
The position of tooltip where it is saying that I dont have proper rights to change the attribute is very weird.
image
The group manager can see member detail too but cannot add groups to the meber, the priviledge exception occurs.

**Comment by xkostka2** [comment]
> > Authorization in Overview, Groups, Applications and Resources pages does not work at all (tested on a member without any admin privileges).
> 
> I am not sure, how did you manage to get to those pages as a member without any privileges. Could you give more information on how to recreate this problem? When I tried to access those pages as a member, it was not possible to get there without manually typing the URL. And even then it worked properly.
> image

When I tried it today, the pages seem to be working correctly. Maybe the problem was on my side, however, I did not change anything.

But still, shouldn't the user be able to see the member attributes page settings when he has privileges to manage them? 

image

**Comment by xkostka2** [comment]
image

image

I also suggest not to hide "Save" and "Remove" buttons when selection is empty, but instead let them disabled.

File: apps/admin-gui/src/app/vos/pages/member-detail-page/member-settings/member-settings-overview/member-settings-overview.component.ts [hunk 1]
```diff
@@ -19,7 +19,8 @@ export class MemberSettingsOverviewComponent implements OnInit {
     private voService: VosManagerService,
     private memberService: MembersService,
     protected route: ActivatedRoute,
-    protected router: Router
+    protected router: Router,
```
Comment (xkureck) [inline apps/admin-gui/src/app/vos/pages/member-detail-page/member-settings/member-settings-overview/member-settings-overview.component.ts:22]: you dont use it anywhere

File: libs/perun/components/src/lib/change-expiration-dialog/change-expiration-dialog.component.ts [hunk 2]
```diff
@@ -44,6 +44,7 @@ export class ChangeExpirationDialogComponent implements OnInit {
     if(!this.expirationAttr){
      this.attributesManagerService.getAttributeDefinitionByName(Urns.MEMBER_DEF_EXPIRATION).subscribe(att => {
        this.expirationAttr = att;
+       console.log(this.expirationAttr.writable);
```
Comment (xkureck) [inline libs/perun/components/src/lib/change-expiration-dialog/change-expiration-dialog.component.ts:47]: I guess this log can be removed

File: libs/perun/components/src/lib/groups-list/groups-list.component.ts [hunk 3]
```diff
@@ -118,6 +126,13 @@ export class GroupsListComponent implements AfterViewInit, OnChanges {
       this.dataSource.paginator = this.paginator;
       this.dataSource.filter = this.filter;
     }
+
+    if(this.vo === undefined && this.groups.length !== 0) {
+      this.vo  = {
+        id: this.groups[0].voId,
+        beanName: "Vo"
+      };
+    }
```
Comment (xkureck) [inline libs/perun/components/src/lib/groups-list/groups-list.component.ts:135]: would probably move this piece of code to some more proper function, or place

**Review by xkureck — CHANGES_REQUESTED** [review]
image
Bug when I click on attributes the settings item in side menu would expand.

**Review by xkureck — APPROVED** [review]

**Pull request merged**